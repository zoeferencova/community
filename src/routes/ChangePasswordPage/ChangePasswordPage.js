import React, { Component } from "react";
import { Link } from "react-router-dom";
import CommUnityContext from "../../contexts/context";
import UserDataService from "../../services/user-data-service";
import AuthApiService from "../../services/auth-api-service";

export default class ChangePasswordPage extends Component {
    static contextType = CommUnityContext;

    state = {
        error: null
    }
    
    handleSubmit = e => {
        e.preventDefault();
        const { old_password, new_password, confirm_password } = e.target;
        const { user } = this.context
        
        if (new_password.value !== confirm_password.value) {
            return this.setState({ error: "New password and confirm password do not match" })
        }

        AuthApiService.checkPassword({ password: old_password.value })
            .then(res => {
                if (res.error) {
                    return this.setState({ error: "Old password is incorrect" })
                } else {
                    return UserDataService.patchUser({ password: new_password.value }, user.id)
                        .then(res => {
                            if (!res.ok) {
                                res.json().then(resJson => this.setState({ error: resJson.error }))
                            } else {
                                console.log("success")
                                this.props.history.push("/account")
                            }
                        })
                }
            })
    }
    
    render() {
        return ( 
            <div>
                <h3>Change Password</h3>
                {this.state.error && 
                <div role="alert">
                    <p>{this.state.error}</p>
                </div>} 
                <form onSubmit={e => this.handleSubmit(e)}>
                    <div>
                        <label htmlFor="old_password">Old Password</label>
                        <input type="password" name="old_password" id="old_password" />
                    </div>
                    <div>
                        <label htmlFor="new_password">New Password</label>
                        <input type="password" name="new_password" id="new_password" />
                    </div>
                    <div>
                        <label htmlFor="confirm_password">Confirm New Password</label>
                        <input type="password" name="confirm_password" id="confirm_password" />
                    </div>
                    <Link to="/account"><button>Cancel</button></Link>
                    <button type="submit">Change Password</button>
                </form>
            </div>  
        )
    }
}