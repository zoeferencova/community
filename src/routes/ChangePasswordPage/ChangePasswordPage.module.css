.main {
    display: flex;
    flex-direction: column;
    align-items: center;
    position: absolute;
    top: 65px;
    width: 90%;
    background-color: white;
    padding: 30px 0;
    box-shadow: 0 1px 3px 0 rgba(0,0,0,.1),0 1px 2px 0 rgba(0,0,0,.06);
    border-radius: 10px;
    margin: 30px auto;
} 

.form {
    width: 75%;
}

.buttonSection {
    width: 100%;
    display: flex;
    justify-content: space-between;
}

.submitButton {
    min-width: 150px;
}

@media only screen and (min-width: 600px) {
    .main {
        width: 80%;
    }
}

@media only screen and (min-width: 800px) {
    .main {
        width: 50%;
    }
}