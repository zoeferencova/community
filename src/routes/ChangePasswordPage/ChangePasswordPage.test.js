import React from "react";
import { render } from "enzyme";
import ChangePasswordPage from "./ChangePasswordPage";
import { BrowserRouter } from "react-router-dom";
import CommUnityContext from "../../contexts/context";

import mockContext from "../../contexts/mock-context";

describe("ChangePasswordPage", () => {
    it("renders ChangePasswordPage component given mock context data", () => {
        const wrapper = render(
            <CommUnityContext.Provider value={mockContext}>
                <BrowserRouter>
                    <ChangePasswordPage />
                </BrowserRouter>
            </CommUnityContext.Provider>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});