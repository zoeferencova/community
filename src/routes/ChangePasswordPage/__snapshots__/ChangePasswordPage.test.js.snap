// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePasswordPage renders ChangePasswordPage component given mock context data 1`] = `
<div
  class="main"
>
  <h3>
    Change Password
  </h3>
  <form
    class="form"
  >
    <div>
      <label
        class="label "
        for="old_password"
      >
        Old Password
      </label>
      <input
        class="input "
        id="old_password"
        name="old_password"
        type="password"
      />
    </div>
    <div>
      <label
        class="label "
        for="new_password"
      >
        New Password
      </label>
      <input
        class="input "
        id="new_password"
        name="new_password"
        type="password"
      />
    </div>
    <div>
      <label
        class="label "
        for="confirm_password"
      >
        Confirm New Password
      </label>
      <input
        class="input "
        id="confirm_password"
        name="confirm_password"
        type="password"
      />
    </div>
    <div
      class="buttonSection"
    >
      <button
        class="button "
        type="button"
      >
        Cancel
      </button>
      <button
        class="buttonDark submitButton"
        loading="false"
        type="submit"
      >
        Change Password
      </button>
    </div>
  </form>
</div>
`;
